pkg/asset/machines/aws/awsmachines.go

```diff
@@ -23,12 +23,13 @@ import (
 
 // MachineInput defines the inputs needed to generate a machine asset.
 type MachineInput struct {
-	Role     string
-	Pool     *types.MachinePool
-	Subnets  map[string]string
-	Tags     capa.Tags
-	PublicIP bool
-	Ignition *capa.Ignition
+	Role           string
+	Pool           *types.MachinePool
+	Subnets        map[string]string
+	Tags           capa.Tags
+	PublicIP       bool
+	PublicIpv4Pool *string
```

Comment:
I don't see why make it a pointer
```suggestion
	PublicIpv4Pool string
```